bin/generate-gitlab-ci-yml.sh

```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env bash
+
+original_directory=$(pwd)
+. "${original_directory}/share/spack/setup-env.sh"
+
+if [ -z "${DOWNSTREAM_CI_REPO}" ] ; then
+    echo "ERROR: missing variable: DOWNSTREAM_CI_REPO" >&2
+    exit 1
+fi
+
+if [ -z "${SPACK_RELEASE_ENVIRONMENT_PATH}" ] ; then
+    echo "ERROR: missing variable: SPACK_RELEASE_ENVIRONMENT_PATH" >&2
+    exit 1
+fi
+
+if [ -z "${CDASH_AUTH_TOKEN}" ] ; then
+    echo "ERROR: missing variable: CDASH_AUTH_TOKEN" >&2
+    exit 1
+fi
+
+if [ -z "${SPACK_RELEASE_ENVIRONMENT_REPO}" ] ; then
+    echo "Assuming spack repo contains environment" >&2
+    env_repo_dir=${original_directory}
+else
+    echo "Cloning environment repo: ${SPACK_RELEASE_ENVIRONMENT_REPO}" >&2
+    mkdir /work
+    cd /work
```

Comment:
This is fine for our CI images, but will likely be a problem for sites running directly on their systems.  Better would be to rely on `mktemp -d` and adding a `trap` line to ensure that the temp dir is cleaned up before the script exits.

Of course, you'll need the Python equivalents to the above (`tempfile` and `atexit` modules) if/once you move this over to a Python implementation.

Reply:
> This is fine for our CI images, but will likely be a problem for sites running directly on their systems. Better would be to rely on `mktemp -d` and adding a `trap` line to ensure that the temp dir is cleaned up before the script exits.

Good suggestion, thanks.  I'll do this on the next pass.

> Of course, you'll need the Python equivalents to the above (`tempfile` and `atexit` modules) if/once you move this over to a Python implementation.

Until you mentioned it here, I hadn't imagined we'd make this a python implementation.  I was considering that for the `rebuild-package.sh` script instead.  In that case, there's a lot of work to do, and it's mostly `spack` commands, so it makes some sense to me do make the whole thing a spack command.  Do you think there's a good argument to be made for doing that here as well?

Reply:
Not particularly, no.  Feel free to ignore my point if that doesn't turn out to be a direction you decide to take this work.

Reply:
> This is fine for our CI images, but will likely be a problem for sites running directly on their systems. Better would be to rely on `mktemp -d` and adding a `trap` line to ensure that the temp dir is cleaned up before the script exits.

@opadron Do you know what signals I should provide to `trap` to make sure the cleanup always happens, no matter how the script exits?

Reply:
```Bash
trap 'rm -rf "$tempdir"' INT TERM QUIT EXIT
```